lucet-runtime/lucet-runtime-internals/src/module/dl.rs

```diff
@@ -117,6 +118,94 @@ impl DlModule {
+            let manifest_page_ptr =
+                (function_manifest.as_ptr() as u64 - manifest_page_offset) as *mut c_void;
+
+            // Because the function manifest may not be page-aligned, the manifest may actually
+            // start far past 0, with pages laid out like something like:
+            //
+            // 0                X                   Y
+            // | data data data | function manifest |
+            //
+            // where Y-X == manifest length as bytes, but X can be anywhere in the PAGE_SIZE bytes
+            // after 0. Since mprotect will start at 0 from whatever address, we must actually
+            // adjust protections for the whole span [0..Y). We do this by specifying that region
+            // as the length to mprotect.
+            let manifest_region_len = (manifest_page_offset
+                + (function_manifest.len() * std::mem::size_of::<FunctionSpec>()) as u64)
+                as usize;
+
+            mprotect(
+                manifest_page_ptr,
+                manifest_region_len,
+                ProtFlags::PROT_READ | ProtFlags::PROT_WRITE,
+            )?;
+
+            for import in module_data.import_functions() {
+                let host_sym = match import.mapped_to {
+                    Some(sym) => sym,
+                    None => {
+                        return Err(lucet_incorrect_module!(
+                            "unbound symbol cannot be resolved: {}/{}",
+                            import.module,
+                            import.name
+                        ));
+                    }
+                };
+
+                let ptr = runtime
+                    .get::<extern "C" fn()>(host_sym.as_bytes())
+                    .map_err(Error::DlError)?;
+
+                let existing_entry = function_manifest
+                    .get_mut(import.fn_idx.as_u32() as usize)
+                    .unwrap();
+
+                // TODO: if this import is ever resolved from another WASM module, we can
+                // read trap and code length information from that module. In fact, this might even
+                // be rewritable as a replace by a clone of the exported FunctionSpec that satisfies
+                // this import!
+                let new_entry = FunctionSpec::new(
+                    *ptr as u64,
+                    existing_entry.code_len(),
+                    existing_entry
+                        .traps()
+                        .map(|manifest| manifest.traps.as_ptr() as u64)
+                        .unwrap_or(0),
+                    existing_entry.traps_len(),
+                );
+
+                std::mem::replace(existing_entry, new_entry);
+            }
+
+            // The function manifest may end up in the same page as code, for now, so this gets +x to
+            // permit those functions to execute. This should be should be fixed so .rodata can be
```

Comment:
typo in comment, also could we make an issue so we don't forget this?

Reply:
:+1: good catch